### PR TITLE
turn SubState into a class, subsequent refactor

### PR DESCRIPTION
**Pull request by gawquon**

### CODE REVIEW

**Review by Zemurin — APPROVED** [review]

**Comment by gawquon** [comment]
Did some reorganization things in each extra commit, let me know if you don't like any of them.

File: EU4toV3/Source/V3World/ClayManager/SubState.cpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+const std::string& V3::SubState::getSourceOwnerTag() const
+{
+	return sourceOwner->getTag();
+}
+
+const std::string& V3::SubState::getHomeStateName() const
+{
+	return state->getName();
+}
```
Comment (Idhrendur) [inline EU4toV3/Source/V3World/ClayManager/SubState.cpp:22]: These are all simple enough to consider putting in the header.

Comment (gawquon) [reply EU4toV3/Source/V3World/ClayManager/SubState.cpp:22]: Can't, circular dependency means I can't call the functions of V3::Country. Can only include the V3::Country .h file in the cop with a forward declaration of the V3::Country in this .h. Any better solution to circular dependency than this?

Comment (Idhrendur) [reply EU4toV3/Source/V3World/ClayManager/SubState.cpp:22]: Ah. Nope, stick with this then.

**Review by Zemurin — APPROVED** [review]

**Pull request merged**